### PR TITLE
M2 3000 mixpanel setup

### PR DESCRIPTION
**Pull request by DimitrySakalouski**

### CODE REVIEW

File: src/shared/utils/mixpanel.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export const Mixpanel = {
+  init() {
+    this.track('INFO', {
+      os: uaParser.getOS().name,
+      width: window.innerWidth,
+      height: window.innerHeight,
+    });
+  },
+  trackPageView(pageName: string) {
+    if (shouldEnableMixpanel) mixpanel.track_pageview({ Admin: pageName });
```
Comment (BamMironov) [inline src/shared/utils/mixpanel.ts:21]: Please correct the passing value to 
```
{ page: '[Admin] ${pageName}' }
```

File: src/shared/utils/mixpanel.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  track(action: string, payload: Dict) {
+    if (shouldEnableMixpanel) mixpanel.track(`[Legacy][Web] ${action}`, payload);
```
Comment (BamMironov) [inline src/shared/utils/mixpanel.ts:24]: Please correct the first argument: `[Admin] ${action}`

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/index.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
+import { Mixpanel } from 'shared/utils';
```
Comment (IhorNesterenko99) [inline src/index.tsx:3]: pls add empty row before

File: src/modules/Auth/state/Auth.reducer.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { AxiosError } from 'axios';
 import { ActionReducerMapBuilder } from '@reduxjs/toolkit';
 
+import { Mixpanel } from 'shared/utils';
```
Comment (IhorNesterenko99) [inline src/modules/Auth/state/Auth.reducer.ts:4]: pls add empty row after

File: src/modules/Auth/features/Login/LoginForm/LoginForm.tsx [hunk 5]
```diff
@@ -33,6 +33,10 @@ export const LoginForm = () => {
     defaultValues: { email: '', password: '' },
   });
 
+  useEffect(() => {
+    Mixpanel.trackPageView('Login');
+  }, []);
```
Comment (BamMironov) [inline src/modules/Auth/features/Login/LoginForm/LoginForm.tsx:38]: All effects should be declared at the end of a function body, just before the `return` statement

File: src/modules/Auth/features/Login/LoginForm/LoginForm.tsx [hunk 6]
```diff
@@ -44,11 +48,20 @@ export const LoginForm = () => {
     if (signIn.fulfilled.match(result)) {
       if (fromUrl) navigate(fromUrl);
       navigateToLibrary(navigate);
+      Mixpanel.track('Login Successful', {});
```
Comment (BamMironov) [inline src/modules/Auth/features/Login/LoginForm/LoginForm.tsx]: No need to pass an empty object as a payload

File: src/modules/Dashboard/features/Applet/AddUser/LinkGenerator/LinkGenerator.tsx [hunk 7]
```diff
@@ -8,6 +8,7 @@ import { useAsync } from 'shared/hooks/useAsync';
 import { StyledFlexTopCenter, StyledTitleBoldMedium, theme } from 'shared/styles';
 import { Tooltip } from 'shared/components';
 
+import { Mixpanel } from 'shared/utils';
```
Comment (BamMironov) [inline src/modules/Dashboard/features/Applet/AddUser/LinkGenerator/LinkGenerator.tsx:11]: Please fix the import styles issues everywhere you made changes. 
As I understand, "shared" imports should be grouped together. And every group should be separated from each other by empty lines.
@IhorNesterenko99 is that correct?

File: src/modules/Auth/features/Login/LoginForm/LoginForm.tsx (hunk 5 above)
Comment (BamMironov) [reply src/modules/Auth/features/Login/LoginForm/LoginForm.tsx:38]: Actually, it would be great if you moved to effect to the page component

**Review by muhurov — APPROVED** [review]

File: src/modules/Auth/features/SignUp/SignUpForm/SignUpForm.tsx [hunk 8]
```diff
@@ -56,13 +56,19 @@ export const SignUpForm = () => {
...
     }
 
     if (signUp.rejected.match(result)) {
       setErrorMessage(getErrorMessage(result.payload));
     }
   };
 
+  useEffect(() => {
+    Mixpanel.trackPageView('Create Account');
+  }, []);
```
Comment (BamMironov) [inline src/modules/Auth/features/SignUp/SignUpForm/SignUpForm.tsx:70]: Please move it to the page component

File: src/modules/Auth/features/Login/LoginForm/LoginForm.tsx [hunk 9]
```diff
@@ -44,11 +48,20 @@ export const LoginForm = () => {
...
       if (fromUrl) navigate(fromUrl);
       navigateToLibrary(navigate);
+      Mixpanel.track('Login Successful');
     }
 
     if (signIn.rejected.match(result)) {
       setErrorMessage(getErrorMessage(result.payload));
     }
+
+    Mixpanel.track('Login Button click');
```
Comment (BamMironov) [inline src/modules/Auth/features/Login/LoginForm/LoginForm.tsx:58]: Every click should be tracked. I believe the `onSubmit` function is called only after the validation is passed.

File: src/modules/Dashboard/pages/Main/Main.tsx [hunk 10]
```diff
@@ -1,12 +1,18 @@
+import { useEffect } from 'react';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/pages/Main/Main.tsx:1]: pls add empty row after importing from react

File: src/modules/Library/features/AppletDetails/AppletDetails.tsx [hunk 11]
```diff
@@ -11,6 +11,7 @@ import { Header, RightButtonType } from 'modules/Library/components';
 import { useAppletsFromCart, useReturnToLibraryPath } from 'modules/Library/hooks';
 import { library } from 'modules/Library/state';
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/AppletDetails/AppletDetails.tsx:13]: remove this line pls and add empty line after

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts [hunk 12]
```diff
@@ -310,6 +311,8 @@ export const useSaveAndPublishSetup = (
     }
   };
   const handleSaveAndPublishFirstClick = async () => {
+    Mixpanel.track('Applet Save click');
```
Comment (BamMironov) [inline src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts:314]: I believe it should be moved to `handleSaveChangesSaveSubmit` instead.

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/shared/components/Password/EnterAppletPassword/EnterAppletPassword.tsx [hunk 13]
```diff
@@ -9,6 +9,7 @@ import { getAppletEncryptionInfo, getParsedEncryptionFromServer } from 'shared/u
 import { Svg, EnterAppletPasswordForm, EnterAppletPasswordProps } from 'shared/components';
 import { useEncryptionStorage } from 'shared/hooks';
 
+import { Mixpanel } from 'shared/utils';
```
Comment (IhorNesterenko99) [inline src/shared/components/Password/EnterAppletPassword/EnterAppletPassword.tsx:12]: remove before and add after empty row

File: src/modules/Dashboard/features/Applet/AddUser/LinkGenerator/LinkGenerator.tsx (hunk 7 above)
Comment (IhorNesterenko99) [reply src/modules/Dashboard/features/Applet/AddUser/LinkGenerator/LinkGenerator.tsx:10]: yes)

**Review by BamMironov — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**